src/main/java/modify_event_use_case/ModifyEventInteractor.java

```diff
@@ -0,0 +1,64 @@
+package modify_event_use_case;
+
+import java.time.LocalTime;
+
+/**
+ * Interactor for the Modify Event use case. Calls the dsGateway and Output Boundary to appropriately execute
+ * application logic.
+ * @author Daniel Livshits
+ */
+public class ModifyEventInteractor implements ModifyEventInputBoundary{
+    private final ModifyEventOutputBoundary outputBoundary;
+    private final ModifyEventDsGateway dsGateway;
+
+    private static final String[] DAYSOFWEEK = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
+
+    public ModifyEventInteractor(ModifyEventOutputBoundary outputBoundary, ModifyEventDsGateway dsGateway){
+        this.outputBoundary = outputBoundary;
+        this.dsGateway = dsGateway;
+    }
+
+    /**
+     * Modifies the desired event by making calls to the outputBoundary and dsGateway, if there are no conflicts.
+     * If there is a naming conflict or time conflict, or if the times are invalid in some way,
+     * creates an instance of Output Data representing the corresponding failure state.
+     *
+     * @param inputData - The data required to modify the desired event.
+     * @return An instance of output data, containing the necessary information about the success or failure of
+     * the modification.
+     */
+    @Override
+    public ModifyEventOutputData modify(ModifyEventInputData inputData){
+        if(!(inputData.getNewStartTime().matches("[01][0-9]:[0-5][0-9]|2[0-3]:[0-5][0-9]"))) {
```

Comment:
One other avenue for debugging is checking whether the title of the event exists within the given day, considering modifying the functionality of your interactor to account for this case, as we are still in an intermediary part of implementing our program.